packages/mobile/src/transactions/send.ts

```diff
@@ -115,7 +118,10 @@ export function* sendTransactionPromises(
     sendTransactionAsync,
     tx,
     account,
-    stableTokenAddress,
+    // Use stableToken to pay fee, unless its balance is Zero
+    // then use Celo (goldToken) to pay fee (pass undefined)
+    // TODO: make it transparent for a user
+    stableTokenBalance.isGreaterThan(0) ? stableTokenAddress : undefined,
```

Comment:
Should we check if the balance is greater than the expected amount + fee instead of just non-zero?

Reply:
well, this case right now is target only for the cases, when user has just created a wallet and transferred Celo from somewhere. Later we would want to do that, when we would also make it transparent to the user.

Reply:
Ok